docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,335 @@
+    )
+```
+
+Now, we have our `@goal_rule`, but we cannot call the `get_project_version_file_view` function; it's Pants that will determine that a rule is used and will make a function call. Well, what should you do to tell Pants you need that rule executed? You should make a function call that:
+* passes an object of the type that matches the type of the rule's input arguments
+* requests an object of the type that a rule returns (you can see that in a type hint)
+
+For this, you can use [`Get`](https://www.pantsbuild.org/docs/rules-api-concepts#await-get---awaiting-results-in-a-rule-body):
+
+```python
+@goal_rule
+async def goal_show_project_version(console: Console, targets: Targets) -> ProjectVersionGoal:
+    targets = [tgt for tgt in targets if tgt.alias == ProjectVersionTarget.alias]
+    for target in targets:
+        project_version = await Get(ProjectVersionFileView, ProjectVersionTarget, target)
+        console.print_stdout(project_version)
+    return ProjectVersionGoal(exit_code=0)
+```
+
+Understanding that calling `Get()` is what causes a particular `@rule` to be executed is essential! It may feel awkward that you cannot run your function. However, by using the `Get()`, you are asking Pants to run your rule, and only knowing this will get you quite far! 
+
+Compare this `Get()` call with the rule signature:
+
+```python
+# requesting an object of type "ProjectVersionFileView", passing an object of type "ProjectVersionTarget" in the variable "target"
+Get(ProjectVersionFileView, ProjectVersionTarget, target)
+```
+
+```python
+# it requires an object of type "ProjectVersionTarget" and will return an object of type "ProjectVersionFileView"
+@rule
+async def get_project_version_file_view(target: ProjectVersionTarget) -> ProjectVersionFileView: ...
+```
+
+The `project-version` Pants goal now shows some useful information -- the target path along with a dummy version. This means our `@rule` was run!
+
+```
+$ ./pants project-version myapp
+ProjectVersionFileView(path='myapp:main-project-version', version='1.2.3')
+```
+
+## Reading the `VERSION` file
+
+Let's read the `VERSION` file and print the version number in the terminal. The `source` field of our target needs to be ["hydrated"](https://www.pantsbuild.org/docs/rules-api-and-target-api#sourcesfield). [Reading a file](https://www.pantsbuild.org/docs/rules-api-file-system) is pretty straightforward as well. We use `Get()` to transform our inputs as needed. Knowing what class you need to request may be tricky, so make sure to review the documentation, and ask for help if you are stuck!    
+
+```python
+@rule
+async def get_project_version_file_view(
+    target: ProjectVersionTarget,
+) -> ProjectVersionFileView:
+    sources = await Get(HydratedSources, HydrateSourcesRequest(target[SourcesField]))
+    digest_contents = await Get(DigestContents, Digest, sources.snapshot.digest)
+    if len(digest_contents) > 1:
+        raise ValueError("There should be only one source file.")
```

Comment:
The source field already does this check.. fwiw

Reply:
Right, since it's the `SingleSourceField`, it guarantees there's only a single file. I've removed the redundant check.